### PR TITLE
Adyen: Pass `subMerchantId` as `additionalData`

### PR DESCRIPTION
**Pull request by naashton**
Payment facilitators must pass this value for auth/purchase transactions

CE-839

Unit:
69 tests, 325 assertions, 0 failures, 0 errors, 0 pendings, 0 omissions, 0 notifications
100% passed

Remote:
91 tests, 351 assertions, 0 failures, 0 errors, 0 pendings, 0 omissions, 0 notifications
100% passed

### CODE REVIEW

**Comment by naashton** [comment]
The random line from `global_collect` was a lint thing.

File: test/unit/gateways/adyen_test.rb [hunk 1]
```diff
@@ -810,6 +810,16 @@ def test_authorize_with_network_tokenization_credit_card
     assert_success response
   end
 
+  def test_authorize_with_sub_merchant_id
```
Comment (therufs) [inline test/unit/gateways/adyen_test.rb:813]: Does this test actually pass `sub_merchant_id`? :confused_england:

Comment (naashton) [reply test/unit/gateways/adyen_test.rb:813]: Hah, whoops. It is not actually testing whether `sub_merchant_id` gets passed.

File: test/unit/gateways/adyen_test.rb [hunk 2]
```diff
@@ -810,6 +810,17 @@ def test_authorize_with_network_tokenization_credit_card
...
+    response = stub_comms do
+      @gateway.authorize(@amount, @credit_card, @options.merge(sub_merchant_id: '12345abcde67890'))
+    end.check_request do |endpoint, data, headers|
+      parsed = JSON.parse(data)
+      puts parsed['additionalData']
```
Comment (therufs) [inline test/unit/gateways/adyen_test.rb:818]: Might be good form to remove this `puts`. Otherwise lgtm!

Comment (naashton) [reply test/unit/gateways/adyen_test.rb:818]: It's just one of those days

Comment (therufs) [reply test/unit/gateways/adyen_test.rb]: That's why we have code review. :D

**Pull request merged**